metacatalog/api/find.py

```diff
@@ -1037,13 +1038,13 @@ def find_entry(session,
         ps = nltk.PorterStemmer()
 
         for key, value in details.items():
-            query = query.filter(models.Detail.stem==ps.stem(key))
+            query = query.filter(_match(models.Detail.stem, ps.stem(key)))
             
             # handle nested json data
             if isinstance(value, (list, tuple, dict)):
                 query = query.filter(models.Detail.raw_value.contains(value))
             else:
-                query = query.filter(models.Detail.raw_value.contains({'__literal__': value}))
+                query = query.filter(_match(models.Detail.raw_value['__literal__'].astext, str(value)))
```

Comment:
I am not so sure here, how SQLAlchemy will translate this into a SQL query. If we do not use the `contains`, the query might be translated to something that will extract the keys as `character varying` and then apply the pattern matching. That might be too slow.

Something we should do (ie. via extra issue):

Use one of the local instances on metacatalog to create a query, but return the `Query` object and translate to SQL:
```python
query = api.find(..., return_iterator=True)
sql = str(query)   
```

Then we can run in pgadmin: `EXPLAIN ANALYZE <sql>`
This gives us some insight into how this is run.

Reply:
API search:
```python 
api.find_entry(session, details={'answer': 42}, return_iterator=True)
```

SQL:
```sql
SELECT entries.id, entries.uuid, entries.title, entries.abstract, entries.external_id, ST_AsEWKB(entries.location) AS location, ST_AsEWKB(entries.geom) AS geom, entries.version, entries.latest_version_id, entries.is_partial, entries.comment, entries.citation, entries.license_id, entries.variable_id, entries.datasource_id, entries.embargo, entries.embargo_end, entries.publication, entries."lastUpdate" 
FROM entries JOIN details ON entries.id = details.entry_id 
WHERE entries.is_partial = false AND entries.latest_version_id IS NULL AND details.stem = 'answer' AND (details.raw_value ->> '__literal__') = '42'
```

This is the important filtering part (from `EXPLAIN ANALYZE`):
`((raw_value ->> '__literal__'::text) = '42'::text)`

I think the `->>` is the correct operator, as it [returns a jsonb object field as text](https://www.postgresql.org/docs/9.3/functions-json.html#:~:text=1%2C2%2C3%5D%27%3A%3Ajson%2D%3E%3E2-,%2D%3E%3E,%27%7B%22a%22%3A1%2C%22b%22%3A2%7D%27%3A%3Ajson%2D%3E%3E%27b%27,-%23%3E), so that we can compare it with the searched value using the `_match()` function and enable wildcard search.